tests/unit/restapi/test_tags_v1.py

```diff
@@ -0,0 +1,525 @@
+
+import pytest
+from flask.testing import FlaskClient
+from flask_sqlalchemy import SQLAlchemy
+from werkzeug.test import TestResponse
+
+from dioptra.restapi.routes import V1_AUTH_ROUTE, V1_TAGS_ROUTE, V1_USERS_ROUTE, V1_ROOT
+
+
+# -- Fixtures --------------------------------------------------------------------------
+
+
+@pytest.fixture
+def auth_account(client: FlaskClient, db: SQLAlchemy) -> dict[str, Any]:
+    user_info = register_user(client)
+    login(client, username=user_info["username"], password=user_info["password"])
+    return user_info
+
+
+@pytest.fixture
+def registered_tags(
+    client: FlaskClient, db: SQLAlchemy, auth_account: dict[str, Any]
+) -> dict[str, Any]:
+    tag1_response = register_tag(
+        client,
+        name="ML: Object Recognition",
+        group_id=auth_account["default_group_id"],
+    )
+    tag2_response = register_tag(
+        client,
+        name="ML: Patch Attack",
+        group_id=auth_account["default_group_id"],
+    )
+    tag3_response = register_tag(
+        client,
+        name="AI",
+        group_id=auth_account["default_group_id"],
+    )
+    return {
+        "tag1": tag1_response,
+        "tag2": tag2_response,
+        "tag3": tag3_response,
+    }
+
+
+# -- Helpers ---------------------------------------------------------------------------
+
+
+def is_iso_format(date_string: str) -> bool:
+    try:
+        datetime.datetime.fromisoformat(date_string)
+        return True
+
+    except ValueError:
+        return False
+
+
+# -- Actions ---------------------------------------------------------------------------
+
+
+def register_user(client: FlaskClient) -> dict[str, Any]:
+    password = "supersecurepassword"
+    response = client.post(
+        f"/{V1_ROOT}/{V1_USERS_ROUTE}",
+        json={
+            "username": "username",
+            "email": "username@example.org",
+            "password": password,
+            "confirmPassword": password,
+        },
+        follow_redirects=True,
+    ).get_json()
+    return {
+        "user_id": response["id"],
+        "username": response["username"],
+        "password": password,
+        "email_address": response["email"],
+        "default_group_id": response["groups"][0]["id"],
+    }
+
+
+def login(client: FlaskClient, username: str, password: str) -> TestResponse:
+    return client.post(
+        f"/{V1_ROOT}/{V1_AUTH_ROUTE}/login",
+        json={"username": username, "password": password},
+    )
+
+
+def register_tag(
+    client: FlaskClient,
+    name: str,
+    group_id: int | None = None,
+) -> TestResponse:
+    """Register a tag using the API.
+
+    Args:
+        client: The Flask test client.
+        name: The name to assign to the new tag.
+        group_id: The group to create the new tag in.
+        description: The description of the new tag.
+
+    Returns:
+        The response from the API.
+    """
+
+    payload = {"name": name}
+
+    if group_id:
+        payload["group_id"] = group_id
```

Comment:
`group_id` is not optional when creating a tag, so I suggest revising this as:

```suggestion
def register_tag(
    client: FlaskClient,
    name: str,
    group_id: int,
) -> TestResponse:
    """Register a tag using the API.

    Args:
        client: The Flask test client.
        name: The name to assign to the new tag.
        group_id: The group to create the new tag in.

    Returns:
        The response from the API.
    """

    payload = {"name": name, "group_id": group_id}
```